pwndbg/dbg/__init__.py

```diff
@@ -30,6 +63,22 @@ def setup(self, *args: Any) -> None:
         """
         raise NotImplementedError()
 
+    def session(self) -> Session | None:
```

Comment:
Why would this ever return `None`?

Reply:
Depends on what _exactly_ we want the session to reflect. LLDB has the notion of an ExecutionContext, which contains information about the currently selected target, process, frame, and thread. It might make sense to have it be `None` in the cases one is not available. It also might not, if we decide that keeping methods in there that don't depend on any of these being available (such as the history method itself).

I feel like it shouldn't hurt to leave that there until we're sure about what we want. Maybe we should add a comment explaining that?

Reply:
It's hard for me to say for sure this early on, but my gut feeling is it would be better to remove this layer of abstraction and if the caller requests something that depends on a non-existent ExecutionContext, we return `None` for those specific function calls. i.e. `dbg.process()` would return `None` if there was no execution context.

Our current usage of `Session` is only `history`, which means we're already using for non-execution context related things. I think it's better to remove it unless we have a concrete need for it. Once we have a concrete need for it, it should be clear at that point exactly what its purpose should be, what methods it should have etc.

Reply:
That makes sense. I'll take it our, then.